resources/views/layout/metadata.blade.php

```diff
@@ -28,6 +28,20 @@
 <meta name="keywords" content="osu, peppy, ouendan, elite, beat, agents, ds, windows, game, taiko, tatsujin, simulator, sim, xna, ddr, beatmania, osu!, osume">
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
+@if ($hasOpengraph)
+    <meta property="og:site_name" content="{{ 'osu! » '.$opghSitename ?? 'osu!' }}">
+    <meta property="og:type" content="website">
+    <meta property="og:url" content="{{ Request::url() }}">
+    <meta property="og:title" content="{{ $opghTitle }}">
+    <meta property="og:description" content="{{ $opghDescription }}">
+
+    @if (isset($opghImage))
+        <meta property="og:image" content="{{ $opghImage }}">
+        <meta property="og:image:width" content="128">
+        <meta property="og:image:height" content="128">
```

Comment:
first these attributes are optional and second nothing on current system has such image size